src/configuration-requestor.spec.ts

```diff
@@ -23,152 +24,174 @@ describe('ConfigurationRequestor', () => {
       sdkName: 'js-client-sdk-common',
       sdkVersion: '1.0.0',
     });
-    const httpClient = new FetchHttpClient(apiEndpoints, 1000);
+    httpClient = new FetchHttpClient(apiEndpoints, 1000);
     flagStore = new MemoryOnlyConfigurationStore<Flag>();
     banditStore = new MemoryOnlyConfigurationStore<BanditParameters>();
     configurationRequestor = new ConfigurationRequestor(httpClient, flagStore, banditStore);
   });
 
+  afterEach(() => {
+    jest.clearAllMocks();
+  });
+
   afterAll(() => {
     jest.restoreAllMocks();
   });
 
-  it('Fetches and stores flag configuration', async () => {
-    const fetchSpy = jest.fn(() => {
-      const response = readMockUFCResponse(MOCK_UFC_RESPONSE_FILE);
-      return Promise.resolve({
-        ok: true,
-        status: 200,
-        json: () => Promise.resolve(response),
-      });
-    }) as jest.Mock;
-    global.fetch = fetchSpy;
-
-    await configurationRequestor.fetchAndStoreConfigurations();
-
-    expect(fetchSpy).toHaveBeenCalledTimes(1); // Flags only; no bandits
-
-    expect(flagStore.getKeys().length).toBeGreaterThanOrEqual(16);
-    const killSwitchFlag = flagStore.get('kill-switch');
-    expect(killSwitchFlag?.key).toBe('kill-switch');
-    expect(killSwitchFlag?.enabled).toBe(true);
-    expect(killSwitchFlag?.variationType).toBe('BOOLEAN');
-    expect(killSwitchFlag?.totalShards).toBe(10000);
-    expect(Object.keys(killSwitchFlag?.variations || {})).toHaveLength(2);
-    expect(killSwitchFlag?.variations['on']).toStrictEqual({
-      key: 'on',
-      value: true,
-    });
-    expect(killSwitchFlag?.variations['off']).toStrictEqual({
-      key: 'off',
-      value: false,
-    });
-    expect(killSwitchFlag?.allocations).toHaveLength(3);
-    const fiftyPlusAllocation = killSwitchFlag?.allocations[1];
-    expect(fiftyPlusAllocation?.key).toBe('on-for-age-50+');
-    expect(fiftyPlusAllocation?.doLog).toBe(true);
-    expect(fiftyPlusAllocation?.rules).toHaveLength(1);
-    expect(fiftyPlusAllocation?.rules?.[0].conditions).toHaveLength(1);
-    expect(fiftyPlusAllocation?.rules?.[0].conditions[0]).toStrictEqual({
-      attribute: 'age',
-      operator: 'GTE',
-      value: 50,
-    });
-    expect(fiftyPlusAllocation?.splits).toHaveLength(1);
-    expect(fiftyPlusAllocation?.splits[0].variationKey).toBe('on');
-    expect(fiftyPlusAllocation?.splits[0].shards).toHaveLength(1);
-    expect(fiftyPlusAllocation?.splits[0].shards[0].salt).toBe('some-salt');
-    expect(fiftyPlusAllocation?.splits[0].shards[0].ranges).toHaveLength(1);
-    expect(fiftyPlusAllocation?.splits[0].shards[0].ranges[0]).toStrictEqual({
-      start: 0,
-      end: 10000,
+  describe('Flags with no bandits', () => {
```

Comment:
Snuck into this PR is not loading bandits if a `null` bandit configuration store is passed and its associated tests.
Most of the below "changes" are just reorganizing the `describe()` blocks, with the main real change being the `Will not fetch bandit parameters if there is no store` test added at the very bottom. Apologies for the clutter.